### PR TITLE
Update executors-overview.md

### PR DESCRIPTION
**Pull request by fritz-astronomer**
Made the executor distinction a little more neutral - corrected some statements

### CODE REVIEW

File: astro/executors-overview.md [hunk 1]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
 
 The Celery executor is the default for all new Deployments. It uses a group of workers, each of which can run multiple tasks at a time. Astronomer uses [worker autoscaling logic](celery-executor.md#celery-worker-autoscaling-logic) to determine how many workers run on your Deployment at a given time.
 
-The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
+The Celery executor is a good option for some use cases. Specifically, the Celery executor is a good fit for your Deployment if:
 
-- You're just getting started with Airflow.
```
Comment (fritz-astronomer) [inline astro/executors-overview.md:28]: Celery has more to configure, and your tasks can crash out - I'd say this is untrue

File: astro/executors-overview.md [hunk 2]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
+    - Celery workers give their tasks 24 hours to finish so they can restart in the event of a code deployment, though this can be alleviated with [DAG Deploys](deploy-dags.md).
 - You want to use different worker types based on the type of task you're running. See [Configure worker queues](configure-worker-queues.md).
-- You have a high number of short-running tasks and want to ensure low latency between tasks.
-- You don't often experience conflicts between Python or OS-level packages and don't require dependency isolation.
+- You have a high number of short-running tasks
+- Your tasks require the shortest startup latency (often milliseconds)
+- You don't require task or dependency isolation
+- You can configure and fine-tune your Celery workers per deployment
 
-If you regularly experience dependency conflicts or find that some tasks consume the resources of other tasks and cause them to fail, Astronomer recommends implementing worker queues or moving to the Kubernetes executor.
```
Comment (fritz-astronomer) [inline astro/executors-overview.md:33]: KubeExec doesn't necessarily help with dependency conflicts

File: astro/executors-overview.md [hunk 3]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
 
-- You have long-running tasks that require more than 24 hours to execute. The Kubernetes executor ensures that tasks longer than 24 hours are not interrupted when you deploy code.
+- You have long-running tasks that may require more than 24 hours to execute.
 - Your tasks are compute-intensive or you are processing large volumes of data within the task. Kubernetes executor tasks run separately in a dedicated Pod per task. 
-- You have a strong understanding of the CPU and memory that your tasks require and would benefit from being able to allocate and optimize infrastructure resources at the task level.
+- Your tasks can tolerate some startup latency (often seconds)
+- Your tasks require task or dependency isolation.
 - You have had issues running certain tasks reliably with the Celery executor.
-  
-The primary limitation with the Kubernetes executor is that each task takes up to 1 minute to start running once scheduled. If you're running short-running tasks and cannot tolerate high latency, Astronomer recommends the Celery executor. To learn more about using the Kubernetes executor, see [Manage the Kubernetes executor](kubernetes-executor.md).
```
Comment (fritz-astronomer) [inline astro/executors-overview.md:50]: I read "up to a minute" as "it takes minutes to launch my task" - it takes seconds, maybe, to launch a task with KE.

**Review by jedcunningham — APPROVED** [review]

File: astro/executors-overview.md [hunk 4]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
+The Kubernetes executor runs each task in an individual container. For each task that needs to run, the executor dynamically launches a Kubernetes Pod on your Astro Cluster for the task. When the task completes, the Pod terminates and its resources are released to the cluster. 
 
-On Astro, the Kubernetes infrastructure required to run the Kubernetes executor is built into every Deployment and is managed by Astronomer. The nodes running Kubernetes Pods are multi-tenant, meaning that Kubernetes Pods on Deployments belonging to different Workspaces can run on the same worker node. This reduces the amount of resources that are required to start up new Pods.
+On Astro, the Kubernetes infrastructure required to run the Kubernetes executor is built into every Deployment and is managed by Astronomer.
 
-The Kubernetes executor is a good fit for teams that want fine-grained control over the execution environment of each of their tasks. Specifically, the Kubernetes executor is a good fit for your Deployment if:
+The Kubernetes executor is a good option for some use cases. Specifically, the Kubernetes executor is a good fit for your Deployment if:
```
Comment (pgagnon) [inline astro/executors-overview.md:46]: I'd avoid repeating "good option for some use cases" in both sections. It seems redundant.

File: astro/executors-overview.md [hunk 5]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
-  
-The primary limitation with the Kubernetes executor is that each task takes up to 1 minute to start running once scheduled. If you're running short-running tasks and cannot tolerate high latency, Astronomer recommends the Celery executor. To learn more about using the Kubernetes executor, see [Manage the Kubernetes executor](kubernetes-executor.md).
+
+You can specify the configuration of the task's Pod, including CPU and memory, as part of your DAG definition using the [Kubernetes Python Client](https://github.com/kubernetes-client/python) and the `pod_override` arg.
+
+If you're running a high task volume or cannot tolerate startup latency, Astronomer recommends the Celery executor. To learn more about using the Kubernetes executor, see [Manage the Kubernetes executor](kubernetes-executor.md).
 
 :::tip
 
 If only some of your tasks need an isolated execution environment, consider using the [KubernetesPodOperator](kubernetespodoperator.md) with the Celery executor.
```
Comment (pgagnon) [inline astro/executors-overview.md:60]: Not part of the PR but I'd avoid framing this as a catch-all suggestion. KPO is more complex than KE in general.

File: astro/executors-overview.md (hunk 4 above)
Comment (fritz-astronomer) [reply astro/executors-overview.md:46]: oh - that was actually intentional 😅 I figured peoples' eyes would glaze over and they'd skip some - so I wanted them to relatively mirror each other. I'll let the writers chew on that one, if they deem me worthy of copy-editing @jwitz @lzdanski

File: astro/executors-overview.md (hunk 5 above)
Comment (cmarteepants) [reply astro/executors-overview.md:60]: IMO, we should remove it. KPOs and KE serve different use-cases

File: astro/executors-overview.md [hunk 6]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
 
 The Celery executor is the default for all new Deployments. It uses a group of workers, each of which can run multiple tasks at a time. Astronomer uses [worker autoscaling logic](celery-executor.md#celery-worker-autoscaling-logic) to determine how many workers run on your Deployment at a given time.
 
-The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
+The Celery executor is a good option for some use cases. Specifically, the Celery executor is a good fit for your Deployment if:
 
-- You're just getting started with Airflow.
+- You do not have tasks that require more than 24 hours to execute
+    - Celery workers give their tasks 24 hours to finish so they can restart in the event of a code deployment, though this can be alleviated with [DAG Deploys](deploy-dags.md).
```
Comment (jwitz) [inline astro/executors-overview.md:29]:
```suggestion
- You don't have tasks that require more than 24 hours to execute. Celery workers give their tasks only 24 hours to finish so they can restart in the event of a code deployment, although you can get around this with [DAG deploys](deploy-dags.md).
```

File: astro/executors-overview.md [hunk 7]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
 - You want to use different worker types based on the type of task you're running. See [Configure worker queues](configure-worker-queues.md).
-- You have a high number of short-running tasks and want to ensure low latency between tasks.
-- You don't often experience conflicts between Python or OS-level packages and don't require dependency isolation.
+- You have a high number of short-running tasks
+- Your tasks require the shortest startup latency (often milliseconds)
+- You don't require task or dependency isolation
+- You can configure and fine-tune your Celery workers per deployment
```
Comment (jwitz) [inline astro/executors-overview.md:34]:
```suggestion
- You have many short-running tasks.
- Your tasks require the shortest startup latency (often milliseconds).
- You don't require task or dependency isolation.
- You want to fine-tune your workers for each Deployment.
```
The last point applies to KE as well right?

File: astro/executors-overview.md (hunk 2 above)
Comment (jwitz) [reply astro/executors-overview.md:33]: Fair enough, I guess you could always use KPO + Celery to achieve the same thing

File: astro/executors-overview.md (hunk 4 above)
Comment (jwitz) [reply astro/executors-overview.md:46]: Hm I think the main issue here is that we no longer have a good "Default" option for someone just getting started. We recommend Celery in our getting started guides, is that still a sound recommendation? If so, we should say something along those lines here as well.

File: astro/executors-overview.md [hunk 8]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
+If you find that some tasks consume the resources of other tasks and cause them to fail, Astronomer recommends implementing worker queues or moving to the Kubernetes executor.
 
 See [Manage the Celery executor](celery-executor.md) to learn more about how to configure the Celery executor.
 
 ### Kubernetes executor
 
-The Kubernetes executor runs each task in an individual Kubernetes Pod instead of in shared Celery workers. For each task that needs to run, the executor calls the Kubernetes API to dynamically launch a Pod for the task. You can specify the configuration of the task's Pod, including CPU and memory, as part of your DAG definition using the [Kubernetes Python Client](https://github.com/kubernetes-client/python) and the `pod_override` arg. When the task completes, the Pod terminates and its resources are restored to the cluster.
+The Kubernetes executor runs each task in an individual container. For each task that needs to run, the executor dynamically launches a Kubernetes Pod on your Astro Cluster for the task. When the task completes, the Pod terminates and its resources are released to the cluster.
```
Comment (jwitz) [inline astro/executors-overview.md:42]:
```suggestion
The Kubernetes executor runs each task in an individual container. For each task that needs to run, the executor dynamically launches a Kubernetes Pod on your Astro cluster to run the task. When the task completes, the Pod terminates and its resources are released to the cluster. 
```

File: astro/executors-overview.md [hunk 9]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
...
 
-On Astro, the Kubernetes infrastructure required to run the Kubernetes executor is built into every Deployment and is managed by Astronomer. The nodes running Kubernetes Pods are multi-tenant, meaning that Kubernetes Pods on Deployments belonging to different Workspaces can run on the same worker node. This reduces the amount of resources that are required to start up new Pods.
+On Astro, the Kubernetes infrastructure required to run the Kubernetes executor is built into every Deployment and is managed by Astronomer.
 
-The Kubernetes executor is a good fit for teams that want fine-grained control over the execution environment of each of their tasks. Specifically, the Kubernetes executor is a good fit for your Deployment if:
+The Kubernetes executor is a good option for some use cases. Specifically, the Kubernetes executor is a good fit for your Deployment if:
 
-- You have long-running tasks that require more than 24 hours to execute. The Kubernetes executor ensures that tasks longer than 24 hours are not interrupted when you deploy code.
+- You have long-running tasks that may require more than 24 hours to execute.
```
Comment (jwitz) [inline astro/executors-overview.md:48]:
```suggestion
- You have long-running tasks that might require more than 24 hours to execute.
```

File: astro/executors-overview.md (hunk 5 above)
Comment (jwitz) [inline astro/executors-overview.md:60]:
```suggestion
The Kubernetes executor is primarily used for resource optimization. If you want more control over the environment and dependencies that your tasks run with, consider using the [KubernetesPodOperator](kubernetespodoperator.md) with either the Celery executor or Kubernetes executor.
```

Comment (jwitz) [reply astro/executors-overview.md:60]: Screenshot 2023-07-14 at 12 24 23 PM
Based on our Learn guide

**Review by jwitz — APPROVED** [review]
Overall LGTM! See my comment on defaults, I think we should still tell users about which executor to pick if they don't care/ just want to get started. If we switch this to KE, we should update our docs ( and maybe product) appropriately.

**Comment by jwitz** [comment]
@fritz-astronomer Can you take one more look at this to check whether my edits work? Then I'll merge!

File: astro/executors-overview.md (hunk 2 above)
Comment (paolaperaza) [reply astro/executors-overview.md:33]: Does it not @fritz-astronomer ?

File: astro/executors-overview.md (hunk 4 above)
Comment (paolaperaza) [reply astro/executors-overview.md:46]: Agreed. I think we have to have a stance on what the default should be, and that's celery right now in our product. A little tricky given limitations but I do think we have to give folks that clarity

File: astro/executors-overview.md (hunk 7 above)
Comment (cmarteepants) [reply astro/executors-overview.md:34]: I wouldn't put that. If you're talking about celery queues, I'd explicitly call that out (it's a differentiator). i.e `You want to direct tasks to run on different celery worker queues. This is useful if you want to run a subset of tasks with priority for SLA reasons, or if your tasks have varied resource requirements.`

File: astro/executors-overview.md (hunk 4 above)
Comment (fritz-astronomer) [reply astro/executors-overview.md:46]: It's a way larger conversation but I think KubeExec is *easier* to start with. Celery has config that you need to think about and the possibility to crash out a bunch of tasks and you won't know why. Kube is just *easier* and "I don't need to think about this" IMO

Comment (fritz-astronomer) [reply astro/executors-overview.md:46]: I feel like celery is default because it was first alphabetically 👀

File: astro/executors-overview.md (hunk 6 above)
Comment (cmarteepants) [reply astro/executors-overview.md:29]: > although you can get around this with [DAG deploys](deploy-dags.md).

Not sure if this is getting in the weeds, but that only helps for restarts due because of dag deploys. It's still an issue for worker scale-downs.

File: astro/executors-overview.md [hunk 10]
```diff
@@ -23,34 +23,40 @@ Read the following topics to learn about the benefits and limitations of each ex
 
 The Celery executor is the default for all new Deployments. It uses a group of workers, each of which can run multiple tasks at a time. Astronomer uses [worker autoscaling logic](celery-executor.md#celery-worker-autoscaling-logic) to determine how many workers run on your Deployment at a given time.
 
-The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
+The Celery executor is a good option for some use cases. Specifically, the Celery executor is a good fit for your Deployment if:
```
Comment (jwitz) [inline astro/executors-overview.md:26]:
```suggestion
The Celery executor is a good option for most use cases and is the default executor for new Deployments. Specifically, the Celery executor is a good fit for your Deployment if:
```

Comment (jwitz) [reply astro/executors-overview.md:26]: I know it's not the right recommendation at the moment, but I don't think product is ready to recommend KE by default yet.

File: astro/executors-overview.md [hunk 11]
```diff
@@ -23,34 +23,39 @@ Read the following topics to learn about the benefits and limitations of each ex
 
 The Celery executor is the default for all new Deployments. It uses a group of workers, each of which can run multiple tasks at a time. Astronomer uses [worker autoscaling logic](celery-executor.md#celery-worker-autoscaling-logic) to determine how many workers run on your Deployment at a given time.
 
-The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
+The Celery executor is a good option for most use cases and is the default executor for new Deployments. Specifically, the Celery executor is a good fit for your Deployment if:
```
Comment (jwitz) [inline astro/executors-overview.md:26]:
```suggestion
The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
```

File: astro/executors-overview.md (hunk 4 above)
Comment (jwitz) [reply astro/executors-overview.md:45]: Final call 🏁 Based on some discussions with product I think we need to still recommend Celery as default for the time being, but we should revisit this in the future.

File: astro/executors-overview.md [hunk 12]
```diff
@@ -25,32 +25,37 @@ The Celery executor is the default for all new Deployments. It uses a group of w
 
 The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
 
-- You're just getting started with Airflow.
+- You don't have tasks that require more than 24 hours to execute. Celery workers give their tasks only 24 hours to finish so they can restart in the event of a code deploy.
```
Comment (jwitz) [inline astro/executors-overview.md:28]:
```suggestion
- You're just getting started with Airflow.
- You don't have tasks that require more than 24 hours to execute. Celery workers give their tasks only 24 hours to finish so they can restart in the event of a code deploy.
```
